Feature/sprint95 update chat sdk doc (#19)
stationfy/arena-chat-sdk: packages/sdk/README.md

To verify if the user was banned, check the property `isBanned` in the user object.

### Q&A

> If the "Q&A" feature is enabled for your chat plan and activate for the current chat, you can implement it using this module. 

Reviewer: We don't need to say this here because users of the SDK are paying customers that will always have the Q&A enaabled.
Suggestion:
"When a Q&A session is enabled for the Chat Room, you can implement the basic Q&A functionality using this module."

First you'll need to get the `qnaProps` with your `chatId` 

```typescript
import Qna from '@/qna/qna.ts'

// get qna props
const qnaProps: QnaProps = await Qna.getQnaProps(chatId)
```

With props in hands, pass it to Qna constructor along with the `qnaId`, the `site` wich is defined in [types](https://github.com/stationfy/arena-chat-sdk/tree/feature/sprint95-update-chat-sdk-doc/packages/types) and the previous defined `arenaChat`

```typescript
// get instance of Qna
const qna = new Qna(qnaProps, qnaId, site, arenaChat)
```

Now is possible to start loading the previously created questions. Just pass a limit of questions to be loaded and the QnaQuestionFilter, wich is also defined in [types](https://github.com/stationfy/arena-chat-sdk/tree/feature/sprint95-update-chat-sdk-doc/packages/types). Both parameters are optional

```typescript
const questions: [QnaQuestion] = await qna.loadQuestions(50, QnaQuestionFilter.RECENT)
```

Start adding new questions just by passing its contents

```typescript
await qna.addQuestion("Which team shall win tonight?")
```

It's also possible to easily awnser a question by calling the following method

```typescript

> const isQuestionAwnsered: Boolean = await qna.awnserQuestion(questions[0], "Lakers should win!")

Reviewer: ```suggestion
const isQuestionAwnsered: Boolean = await qna.awnserQuestion(questionId, "Lakers should win!")
```

```

> Now there are some event listeners that should be noticed

Reviewer: ```suggestion
To listen to changes to questions in real-time, some listeners can be used:
```

- onChange: This will watch for Q&A props changes coming from dashboard and then call the passed callback with the Qna instance updating it properties
```typescript
onChange(callback: (instance: BaseQna) => void): void
```

> - onQuestionReceived: Watchs for new questions updating the question cache and calls the passed callback with the new question

Reviewer: ```suggestion
- onQuestionReceived: Watches for new questions updating the question cache and calls the passed callback with the new question
```

```typescript
onQuestionReceived(callback: (question: QnaQuestion) => void): void
```

### Direct Messages 

> By having the "Private Messages" option enabled in your chat, it's possible to create private dialogs between two or more users.

Reviewer: When you enable the "Private Messages" option in you chat, members can chat directly with each other.

```

In the same way we can list only the private channels that are not group channels for the current user

```typescript
const userPrivateChannels: BasePrivateChannel[] = await arenaChat.getUserPrivateChannel()
```

Also, it's possible to create a channel via `ArenaChat` instance or return an existing one for a userId

```typescript
const bobChannel: BasePrivateChannel = await arenaChat.createUserPrivateChannel(
  "bobUserId",
  new ChatMessageContent(messageParams) // optional
)
```

About the event listeners this one deserves attention, types are defined here [types]('https://github.com/stationfy/arena-chat-sdk/tree/feature/sprint95-update-chat-sdk-doc/packages/types')

- onUnreadMessagesCountChanged : Watchs for unread messages within a private channel so you can display it for the user in real time, callback will hold the unread count

```typescript
onUnreadMessagesCountChanged(user: ExternalUser, site: Site, callback: (total: number) => void): () => void 
```

- offUnreadMessagesCountChanged : From the sdk you can unsubscribe the a previous setted listener for `onUnreadMessagesCountChanged`

```typescript
arenaChat.offUnreadMessagesCountChanged()
```

### Polls

> Adding polls to engage the user experience is quite simple since it's enabled by your chat

Reviewer: ```suggestion
Adding polls to engage the user experience is quite simple since it's enabled by default in all chats
```

> To configure a polls manager you need your previously configured `chatRoom` and a `arenaChat`, refers to [types](https://github.com/stationfy/arena-chat-sdk/tree/feature/sprint95-update-chat-sdk-doc/packages/types)

Reviewer: ```suggestion
To configure a polls manager you need your previously configured `chatRoom` and a `arenaChat`, please refer to [types](https://github.com/stationfy/arena-chat-sdk/tree/feature/sprint95-update-chat-sdk-doc/packages/types)
```

```typescript
import Polls from '@/polls/polls.ts'

const polls: BasePolls = new Polls(chatRoom, arenaChat)
```

Once you get a polls instance it's possible to start loading the existing polls

```typescript
const pollsList: [Poll] = await polls.loadPolls(

>   PollFilter.RECENT, // optional

Reviewer: let's detail what options are available and the meaning of each of them.

  50 // optional
)
```

To register a vote in a option for a poll you need to inform the `pollId` the `optionId` that is a number starting in 0 and optionaly an `anonymousId`

```typescript

> await polls.pollVote(

Reviewer: @cesarwbr i think passing the anonymousId here can be confusing for the sdk users. What happens if the user is already set?

Reviewer: task created: https://app.clickup.com/t/fuuzhf

  pollsList[0].pollId,
  5, // option 5
  "anonUser00023"
)
```

Within a channel context is possible to receive the polls manager so it can be presented to the current user

```typescript
const polls: BasePolls = await channel.getPollsInstance(userId)
```

> About the event listeners the following one is interesting

Reviewer: ```suggestion
Event listeners are exposed to watch for changes in real-time:
```
